rules/no-deferred.js

```diff
@@ -1,16 +1,14 @@
 'use strict'
 
-module.exports = function(context) {
-  function enforce(node) {
-    if (node.callee.type !== 'MemberExpression') return
-    if (node.callee.object.name !== '$') return
-    if (node.callee.property.name !== 'Deferred') return
+const utils = require('./utils.js')
 
+module.exports = function(context) {
+  const enforce = utils.withProperty('Deferred', function(node) {
```

Comment:
Since these helper functions return a function, you can assign it to a variable and use it as a visitor callback for multiple node types (like the `CallExpression` and `NewExpression` in this example).